src/providers/postgres/qgspostgresfeatureiterator.cpp

```diff
@@ -111,12 +112,34 @@ QgsPostgresFeatureIterator::QgsPostgresFeatureIterator( QgsPostgresFeatureSource
     }
   }
 
-  bool success = declareCursor( whereClause, limitAtProvider ? mRequest.limit() : -1, false );
+  QStringList orderByParts;
+
+  mOrderByCompiled = true;
+
+  Q_FOREACH ( const QgsFeatureRequest::OrderByClause& clause, request.orderBys() )
+  {
+    QgsPostgresExpressionCompiler compiler = QgsPostgresExpressionCompiler( source );
+    QgsExpression expression = clause.expression();
+    if ( compiler.compile( &expression ) == QgsSqlExpressionCompiler::Complete )
+    {
+      QString part;
+      part = compiler.result();
+      part += clause.ascending() ? " ASC" : " DESC";
+      part += clause.nullsFirst() ? " NULLS FIRST" : " NULLS LAST";
+      orderByParts << part;
+    }
+    else
+    {
+      mOrderByCompiled = false;
```

Comment:
should this also clear orderByParts? is it an issue if a partial order by clause is passed, or is that desirable?